packages/zero-runtime/src/styled.spec.tsx

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable material-ui/no-empty-box */
+import * as React from 'react';
+import { styled } from '@mui/zero-runtime';
+
+declare module '@mui/zero-runtime/theme' {
+  interface ThemeArgs {
+    theme: {
+      palette: {
+        primary: Record<string, string>;
+      };
+      mixins: {
+        toolbar: Record<string, string>;
+      };
+      typography: {
+        body1: Record<string, string>;
+      };
+    };
+  }
+}
+
+const B = styled('a')<{ isRed?: boolean }>(({ theme }) => ({
+  color: (props) => (props.isRed ? 'red' : 'blue'),
+  backgroundColor: [theme.palette.primary.main, theme.palette.primary.darkMain],
+  '&:enabled': {
+    color: 'beige',
+  },
+  '@layer utils': {},
+  '@media (min-width:360px)': {
+    '.globalClass': {
+      color: [theme.palette.primary.main, theme.palette.primary.darkMain],
+    },
+  },
+  variants: [
+    {
+      props: {
+        isRed: true,
+      },
+      style: {
+        color: ({ isRed }) => (isRed ? 'red' : 'blue'),
+      },
+    },
+    {
+      props: {
+        isBlue: true,
```

Comment:
Shouldn't this throw, as `isBlue` is not part of the props interface?

Reply:
It does not explicitly forbid passing unknown props but'll provide autocompletion for known props.
I [verified](https://codesandbox.io/s/throbbing-brook-4jtv5l?file=/src/App.tsx) that it works the same way with `@mui/system`.

Reply:
In that case, we should fix it in the `@mui/system`, too, and not replicate the wrong behavior (at least in my opinion is wrong). Is there a limitation that it was implemented like this? Can we fix it?

Reply:
I am approving we do a follow up, to allow @siriwatknp to use the types we have so far.

Reply:
I am still exploring a way to fix it. There's one more feature to be added (automatic prop suggestion when `as` prop is provided with a different element. Will try to add both the fixes together.